lib/datasets.go

```diff
@@ -412,7 +445,7 @@ func (r *DatasetRequests) SetPublishStatus(p *SetPublishStatusParams, publishedR
 
 	if p.UpdateRegistry && r.node.Repo.Registry() != nil {
 		var done bool
-		rr := NewRegistryRequests(r.node, nil)
+		rr := registryMethods{node: r.node}
```

Comment:
Stream of conscious here...

If these *Methods are wrapping a single object (as mentioned above), this will be easier to do. Can we use the interface returning function instead of the internal one? Or would it be doable if we switched to this single object. I'm picturing this:

```
state := NewMethodState(node, rpc.client, config)
m := NewDatasetMethods(state)
m.SetPublishStatus(...)
...
func (m *datasetOperations) SetPublishStatus(...) {
  ...
  r := NewRegistryMethods(m.state)
}
```

I guess the problem is that we want NewDatasetMethods and NewRegistryMethods to be methods on the top-level Lib object, right? In that case I guess the question becomes whether `state` can implement the Instance interface.